lib/initialForm/CountrySelectorWidget.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:provider/provider.dart';
+
+class CountrySelectorWidget extends StatefulWidget {
+  final String text;
+  final String disclaimerText;
+
+  const CountrySelectorWidget(
+      {Key? key, required this.text, required this.disclaimerText})
+      : super(key: key);
+
+  @override
+  _CountrySelectorWidgetState createState() => _CountrySelectorWidgetState();
+}
+
+class _CountrySelectorWidgetState extends State<CountrySelectorWidget> {
+  String? _selectedCountryCode = 'IL';
+  bool isvisible = false;
+  void saveLocation(String location, UserInformation userInfo) async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    prefs.setString("location", location);
+    userInfo.updateLocation(location);
+  }
+
+  void changeVisible() {
+    print("pressed");
+    setState(() {
+      isvisible = !isvisible;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final userInfoProvider = Provider.of<UserInformation>(context);
+    final appLocale = AppLocalizations.of(context);
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        ConstrainedBox(
+          constraints: BoxConstraints(maxWidth: 300.w),
+          child: Row(
+            children: [
+              myAutoSizedText(
+                  widget.text,
+                  TextStyle(
+                      fontSize: 18.sp,
+                      fontWeight: FontWeight.normal,
+                      color: Colors.black),
+                  null,
+                  24),
+              TextButton(
+                  style: ButtonStyle(
+                      padding: WidgetStatePropertyAll(
+                          EdgeInsets.symmetric(horizontal: 8.w)),
+                      iconColor: WidgetStatePropertyAll(
+                        Colors.black,
+                      ),
+                      overlayColor: WidgetStatePropertyAll(backgroundGray)),
+                  onPressed: changeVisible,
+                  child: Icon(
+                    Icons.question_mark,
+                    size: 12.sp,
+                  ))
+            ],
+          ),
+        ),
+        CountryCodePicker(
+          onChanged: (country) {
+            setState(() {
+              _selectedCountryCode = country.code;
+              saveLocation(country.code!, userInfoProvider);
+            });
+          },
+          initialSelection: userInfoProvider.location.isNotEmpty
+              ? userInfoProvider.location
+              : 'IL',
+          // favorite: ['+972', 'IL'],
+          showCountryOnly: true,
+          showOnlyCountryWhenClosed: true,
+          alignLeft: false,
+          countryFilter: ['IL', 'US'],
+        ),
+        Container(
+          padding: EdgeInsets.fromLTRB(
+              appLocale!.textDirection == "rtl" ? 0 : 0.w,
```

Comment:
consider:
taking text direction from `locale`
having the numbers as meaningful constants